smtp/message.js

```diff
@@ -12,6 +11,33 @@ const BUFFERSIZE = MIMECHUNK * 24 * 7; // size of the message stream buffer
 
 let counter = 0;
 
+function getRFC2822Date(date = new Date(), useUtc = false) {
```

Comment:
what about pulling these out into a utility file like `RFC2822.js`?

i don't feel too strongly about it, but since you are exporting these functions, it would provide an opportunity to extract them an expose the functions on a file that is more granular focused.

Reply:
I moved these to a new file, `./smtp/date.js`. also moved the tests to a new file & split them by purpose (UTC vs. non-UTC).